Source/ImPlotWidgets/Private/LineFilledGraphWidget.cpp

```diff
@@ -17,24 +19,40 @@ void SFilledGraphWidget::Construct(const FArguments& InArgs)
 			.OnImGuiDrawDelegate(this, &SFilledGraphWidget::OnImGuiDraw)
 		];
 
-	HandleDataArray();
+	HandleDataArrays();
 }
 
-void SFilledGraphWidget::HandleDataArray()
+void SFilledGraphWidget::HandleDataArrays()
 {
-	Algo::Sort(InData2D, [](const FDataPoint& LeftDataPoint, const FDataPoint& RightDataPoint)
-		{
-			return LeftDataPoint.X < RightDataPoint.X;
-		});
+	XMaxValue = DataArrays[0][0].X;
+	YMaxValue = DataArrays[0][0].Y;
+	YMinValue = 0.0f;
 
-	for (FDataPoint Point : InData2D)
+	for (TArray<FDataPoint> CurrentDataArray : DataArrays)
 	{
-		XValues.Add(Point.X);
-		YValues.Add(Point.Y);
-	}
+		Algo::Sort(CurrentDataArray, [](const FDataPoint& LeftDataPoint, const FDataPoint& RightDataPoint)
+			{
+				return LeftDataPoint.X < RightDataPoint.X;
+			});
+
+		TArray<float> XValues;
+		TArray<float> YValues;
+
+		for (FDataPoint Point : CurrentDataArray)
+		{
+			XValues.Add(Point.X);
+			YValues.Add(Point.Y);
+		}
+
+		XValueArrays.Add(XValues);
+		YValueArrays.Add(YValues);
+
+		XMaxValue = (FMath::Max(XValues) > XMaxValue) ? FMath::Max(XValues) : XMaxValue;
+		YMaxValue = (FMath::Max(YValues) > YMaxValue) ? FMath::Max(YValues) : YMaxValue;
```

Comment:
There's an FMath::Max function that handles two values, so you should be able to condense this down a bit
```suggestion
		YMaxValue = FMath::Max(FMath::Max(YValues), YMaxValue);
```

Reply:
Also if you're going to MoveTemp (which you should) these lines need to go before the MoveTemp call. Move operations invalidate the original variable as they transfer ownership of the memory owned by the first variable to the second.